PR: Add FAQ section to docs (#174)
spyder-ide/spyder-docs: doc/faq.rst

   .. image:: images/faq/faq-navigator-install.png
      :alt: Navigator showing installing specific version of Spyder

==============
Running Spyder
==============

.. dropdown:: Q: How do I run Spyder?

   You can launch it in any of the following ways:

   * **From the command line**: Type ``spyder`` in your terminal (or Anaconda prompt on Windows).

   * **From Anaconda Navigator**: Scroll to :guilabel:`Spyder` under :guilabel:`Home`, and click :guilabel:`Launch`.

     .. image:: images/faq/faq-launch-anaconda.png
        :alt: Navigator showing running a specific version of Spyder

   * ***Windows Only***: Launch it via the Start menu shortcut.

     .. image:: images/faq/faq-windows-launch.png
        :alt: Spyder shortcut in the Windows Start menu

.. dropdown:: Q: Can I try Spyder without installing it?

   Yes!
   With `Binder`_, you can work with a fully functional copy of Spyder that runs right in your web browser.
   Try it `here`_.

   .. _Binder: https://mybinder.org/
   .. _here: https://mybinder.org/v2/gh/spyder-ide/spyder/4.x?urlpath=/desktop

.. dropdown:: Q: What are the system requirements for Spyder? How resource-intensive is it?

>    Spyder works on modern versions of Windows, macOS and Linux (see table below for recommended versions) via Anaconda, as well as other methods.

Reviewer: Well, it sounds better to a non-native speaker like me.

   Select the environment you want to launch Spyder from under :guilabel:`Applications on`.
   If Spyder is installed in this environment, you will see it on Navigator's :guilabel:`Home` window.
   Click :guilabel:`Launch` to use Spyder with your selected environment.

   .. image:: images/faq/faq-run-environment.png
      :alt: Navigator showing running Spyder in a specific environment

.. dropdown:: Q: How do I run Spyder in a conda environment using the command line?

   Activate your conda environment typing on your command line (or Anaconda Prompt on Windows):

   .. code-block:: bash

      conda activate <ENVIRONMENT-NAME>

   Then, type ``spyder``.

============
Using Spyder
============

.. dropdown:: Q: How do I install Python packages to use within Spyder?

   The first approach for installing a package should be using conda.
   In your command line (or Anaconda Prompt on Windows) type:

   .. code-block:: bash

      conda install <PACKAGE-NAME>

   If your installation is not successful go through steps 3 to 5 of Part 2 in our `video`_ on solving and avoiding problems with pip, Conda and Conda-Forge.

>    .. _video: https://www.youtube.com/watch?v=Ul79ihg41Rs&t=306s

Author: Not sure about this. Let's see what @ccordoba12 thinks

>    .. _video: https://www.youtube.com/watch?v=Ul79ihg41Rs&t=306s

Reviewer 2: Since the sections are all collapsible, I'm not sure what possible downside you see to embedding it directly so users can view it more easily, rather than having to click through a link (which can sharply reduce the rate at which they do so and takes them away from our site), but we'll see what Carlos thinks.

>    .. _video: https://www.youtube.com/watch?v=Ul79ihg41Rs&t=306s

Reviewer: I agree with @CAM-Gerlach. Let's embed it here and make it start at the right time.

>    .. _video: https://www.youtube.com/watch?v=Ul79ihg41Rs&t=306s

Reviewer 2: Implemented

.. dropdown:: Q: How do I get Spyder to work with my existing Python packages/environment?

>    To work with an existing environment in Spyder, change Spyder’s default python interpreter.

Reviewer: ```suggestion
   To work with an existing environment in Spyder, change Spyder’s default Python interpreter.
```

   To do so, first go to your terminal, type ``conda info --envs``, and copy the path from the environment you created to your clipboard.

   .. image:: images/faq/faq-conda-info.png
      :alt: Navigator showing installing specific version of Spyder

   Now, go to :guilabel:`Preferences` in Spyder’s main window, click :guilabel:`Python interpreter`, check :guilabel:`Use the following Python interpreter` paste the path and add ``/bin/python`` at the end for Mac and Linux or ``/python.exe`` in Windows.

   .. image:: images/faq/faq-python-interpreter.png
      :alt: Navigator showing installing specific version of Spyder

>    Restart Spyder for these changes to take effect.

Author: Not sure what this means?

Reviewer 2: Sorry about that, silly copy/paste mistake on my part. What I meant to say is

```suggestion
   Click :guilabel:`Restart kernel` in the :guilabel:`Consoles` menu for this change to take effect.
```

Much quicker and more accurate, and also describe the change in the proper (singular) tense.

>    Click :guilabel:`Restart kernel` in the :guilabel:`Consoles` menu for this change to take effect.

Reviewer: I think an screenshot is missing at this point, i.e. one that shows where the Consoles menu is placed and what's the `Restart kernel` entry you need to click on it.

Reviewer: It's up to Juanita then. If she thinks this can be understood without a screenshot, I'm fine with it.

Author: I think as it is right now we don't need a screenshot, if we want to change it to be from the console pane then it makes sense.

   Under :guilabel:`General` in Spyder's :guilabel:`Preferences`, go to the :guilabel:`Advanced settings` tab and select your language from the options displayed under :guilabel:`Language`.

   .. image:: images/faq/faq-change-language.png
      :alt: Spyder change language in preferences.

.. dropdown:: Q: How do I use code cells in Spyder?

   To create a cell in Spyder's :doc:`editor`, type ``#%%`` in your script.
   Each ``#%%`` will make a new cell.
   To run a cell, press :kbd:`Shift-Enter` (while your cursor is focused on it) or use the :guilabel:`Run current cell` button in Spyder's toolbar.

   .. image:: images/faq/faq-cells.png
      :alt: Spyder showing cell generation.

.. dropdown:: Q: How do I use plugins with Spyder (e.g. Spyder-Notebook, Spyder-Terminal, Spyder-Unittest)?

   Spyder plugins are available in the ``spyder-ide`` conda channel.
   To install one, type on the command line (or Anaconda Prompt on Windows):

   .. code-block:: bash

      conda install -c spyder-ide <PLUGIN>

   Replace ``<PLUGIN>`` with the name of the plugin you want to use.
   For more information on a specific plugin, go to the its repository:

>    * `spyder-reports`_

Reviewer: Spyder-reports is broken at the moment. Please remove it from this list.

   * `spyder-unittest`_
   * `spyder-terminal`_
   * `spyder-notebook`_
   * `spyder-memory-profiler`_
   * `spyder-line-profiler`_
   * `spyder-vim`_

>    * `spyder-autopep8`_

Reviewer: These two have no official releases yet, so please remove them too from here.

Reviewer 2: Removed

   .. _spyder-line-profiler: https://github.com/spyder-ide/spyder-line-profiler
   .. _spyder-vim: https://github.com/spyder-ide/spyder-vim
   .. _spyder-autopep8: https://github.com/spyder-ide/spyder-autopep8

.. dropdown:: Q: How do I clear all my variables before executing my code?

   Check the option :guilabel:`Remove all variables before execution` in the :guilabel:`Configuration per file...` dialog under :guilabel:`Run` in the Menu bar.

   .. image:: images/faq/faq-remove-variables.png
      :alt: Spyder showing cell generation.

.. dropdown:: Q: How do I run my code in a dedicated console or an external system terminal?

   Select the option in the :guilabel:`Configuration per file...` dialog under :guilabel:`Run` in the Menu bar.

   .. image:: images/faq/faq-run-options.png
      :alt: Spyder showing cell generation.

.. dropdown:: Q: How do I change the syntax highlighting theme?

   Go to :guilabel:`Preferences` and select the theme under :guilabel:`Syntax highlighting theme` in the :guilabel:`Appearance` section.

   .. image:: images/faq/faq-highlighting-theme.png
      :alt: Spyder showing cell generation.

===============

> Troubleshooting

Author: Yes. I wrote that when I opened the PR

> Troubleshooting

Reviewer 2: Sorry, I didn't see it below the screenshot.

> .. dropdown:: Q: How do I change the syntax highlighting theme?

Reviewer: ```suggestion
.. dropdown:: Q: How do I change the syntax highlighting coloring in the editor?
```

* I thought we had agreed on using coloring instead of theme for this question.
* People usually asks for the editor.

Reviewer 2: ```suggestion
.. dropdown:: Q: How do I change the syntax highlighting theme in the Editor?
```

I agree with you about adding the Editor.

As I recall, we went back and forth on this point and @juanis2112 initially sided with you, but then ended up agreeing on syntax highlighting theme. Its the [accepted, understood term](https://en.wikipedia.org/wiki/Syntax_highlighting) for this in English, and is what we ourselves call it in the preferences.

Reviewer: Ok, no problem.

===============

.. dropdown:: Q: I've found a bug or issue with Spyder, what do I do?

   You should first follow the steps in our Troubleshooting guide.
   If you can't solve your bug, open an issue by following the instructions in our Submit a Report section.

.. dropdown:: Q: I get an error in the IPython console running my code! Help!

   First, make sure the error you are having is not an error with your code.

>    For this, try running it in any standard Python interpreter.

Author: I suggest removing the last part you added as the next sentence starts with basically the same thing. @CAM-Gerlach

>    For this, try running it in any standard Python interpreter.

Reviewer 2: Sure, done. Good catch!

>    For this, try running it in any standard Python interpreter.

Author: ```suggestion
   To confirm this, try running it in any standard Python interpreter.
```

   If it turns out the error is with your code, `Stack Overflow`_ might be the best place to start.
   Otherwise, start at the Basic First Aid section of our troubleshooting guide.

   .. _Stack Overflow: https://stackoverflow.com

.. dropdown:: Q: Code completion/help doesn't work; what can I do?

   If nothing is displayed in the calltip, hover hint or help pane, make sure the object you are inspecting has a docstring and try executing your code in the :doc:`ipythonconsole` to get help and completions on the object there.
   If this doesn't work, try restarting the LSP by right-clicking it in the bottom of Spyder's main window and selecting the :guilabel:`Restart Python Language Server` item.

>    For more information go to the Completion/help not working component in the Common Illnesses section of our troubleshooting guide.

Reviewer 2: ```suggestion
   For more information, go to the Completion/help not working section in the Common Illnesses page of our troubleshooting guide.
```

Use more appropriate terms

> .. dropdown:: Q: I get the message "An error occurred while starting the kernel". How do I fix it?

Reviewer 2: ```suggestion
.. dropdown:: Q: I get the message "An error occurred while starting the kernel". How do I fix this?
```

Grammar

>    First make sure your version of spyder-kernels is compatible with your version of Spyder.

Reviewer 2: ```suggestion
   First, make sure your version of Spyder-Kernels is compatible with that of Spyder.
```

Comma, caps and avoid reptition

    .. table::

       ==============   ==============
       Spyder           Spyder-Kernels
       ==============   ==============
       4.0.0-4.0.1      1.8.1
       4.1.0-4.1.2      1.9.0
       4.1.3            1.9.1
       4.1.4            1.9.2
       ==============   ==============

>    To install the right version, type on your command line (or Anaconda Prompt on Windows)

Reviewer 2: ```suggestion
   To install the right version, type the following on the command line (or Anaconda Prompt on Windows)
```

Avoid grammar issues and ambiguous phrasing

   .. code-block:: bash

      conda install spyder-kernels=<VERSION>

>    For more information go to the Errors starting the kernel component in the Common Illnesses section of our troubleshooting guide.

Reviewer 2: ```suggestion
   For more information, go to the Errors starting the kernel section in the Common Illnesses page of our troubleshooting guide.
```

Use more appropriate terms

>        4.1.4            1.9.2

Reviewer: ```suggestion
       4.1.4            1.9.3
```

Reviewer 2: My suggestion must not have gotten applied

>    For more information, go to the Errors starting the kernel section in the :doc:`common-illnesses` page of our troubleshooting guide.

Reviewer: ```suggestion
   For more information, go to the *Errors starting the kernel* section in the :doc:`common-illnesses` page of our troubleshooting guide.
```

It looks odd to me without italics or quotes, to signal that that part makes reference to a header.

Reviewer 2: Instead of doing this, we should directly link the section itself. I'll include a section in my review to this effect.

Reviewer 2: ```suggestion
   For more information, go to the :ref:`starting-kernel-problems-ref` section in the :doc:`common-illnesses` page of our troubleshooting guide.
```

As before, above this section title in the troubleshooting guide, you need to insert:

```
.. _starting-kernel-problems-ref:
```

Reviewer: Even better.

Reviewer 2: Great!

Reviewer 2: Implemented

   If nothing is displayed in the calltip, hover hint or :doc:`help` pane, make sure the object you are inspecting has a docstring, and try executing your code in the :doc:`ipythonconsole` to get help and completions there.
   If this doesn't work, try restarting PyLS by right-clicking the :guilabel:`LSP Python` label item in the statusbar at the bottom of Spyder's main window, and selecting the :guilabel:`Restart Python Language Server` option.

>    For more information, go to the Completion/help not working section in the :doc:`common-illnesses` page of our troubleshooting guide.

Reviewer 2: ```suggestion
   For more information, go to the :ref:`code-completion-problems-ref` section in the :doc:`common-illnesses` page of our troubleshooting guide.
```

Above this section title in the troubleshooting guide, you need to insert:

```
.. _code-completion-problems-ref:
```

Reviewer 2: Implemented
